src/pages/resultsView/survival/SurvivalChart.spec.tsx

```diff
@@ -0,0 +1,184 @@
+import {assert} from 'chai';
+import {PatientSurvival} from "../../../shared/model/PatientSurvival";
+import {
+    getEstimates, getMedian, getLineData, getScatterData, getScatterDataWithOpacity, getStats, calculateLogRank,
+    getDownloadContent, convertScatterDataToDownloadData
+} from "./SurvivalChart";
+
+const exampleAlteredPatientSurvivals = [
+    {
+        patientId: "TCGA-OR-A5J1",
+        studyId: "acc_tcga",
+        months: 5.09,
+        status: false
+    },
+    {
+        patientId: "TCGA-OR-A5J2",
+        studyId: "acc_tcga",
+        months: 0.09,
+        status: true
+    },
+];
+
+const exampleUnalteredPatientSurvivals = [
+    
+    {
+        patientId: "TCGA-OR-A5J3",
+        studyId: "acc_tcga",
+        months: 0,
+        status: false
+    },
+    {
+        patientId: "TCGA-2F-A9KO",
+        studyId: "blca_tcga",
+        months: 63.83,
+        status: true
+    },
+    {
+        patientId: "TCGA-2F-A9KP",
+        studyId: "blca_tcga",
+        months: 0.13,
+        status: false
+    },
+    {
+        patientId: "TCGA-2F-A9KQ",
+        studyId: "blca_tcga",
+        months: 182.19,
+        status: true
+    }
+];
+
+const examplePatientSurvivals = exampleAlteredPatientSurvivals.concat(exampleUnalteredPatientSurvivals);
+
+const exampleAlteredEstimates = [1,0.8];
+const exampleUnalteredEstimates = [0.8,0.5333333333333333,0.5333333333333333,0];
+const exampleEstimates = exampleAlteredEstimates.concat(exampleUnalteredEstimates);
+
+describe("SurvivalChart", () => {
+    describe("#getEstimates()", () => {
+        it("returns empty list for empty list", () => {
+            assert.deepEqual(getEstimates([]), []);
+        });
+
+        it("returns correct estimates for the example data", () => {
+            assert.deepEqual(getEstimates(examplePatientSurvivals), exampleEstimates);
+        });
+    });
+
+    describe("#getMedian()", () => {
+        it("returns NA for empty list", () => {
+            assert.equal(getMedian([], []), "NA");
+        });
+
+        it("returns 182.19 for the example data", () => {
+            assert.equal(getMedian(examplePatientSurvivals, exampleEstimates), "182.19");
+        });
+    });
+
+    describe("#getLineData()", () => {
+        it("returns [{x: 0, y: 100}] for empty list", () => {
+            assert.deepEqual(getLineData([], []), [{x: 0, y: 100}]);
+        });
+
+        it("returns correct line data for the example data", () => {
+            assert.deepEqual(getLineData(examplePatientSurvivals, exampleEstimates), [
+                {x: 0, y: 100},
+                {x: 5.09, y: 100},
+                {x: 0.09, y: 80},
+                {x: 0, y: 80},
+                {x: 63.83, y: 53.3333333333333333},
+                {x: 0.13, y: 53.3333333333333333},
+                {x: 182.19, y: 0}
+            ]);
+        });
+    });
+
+    describe("#getScatterData()", () => {
```

Comment:
this is a function which arguably has no logic inside it.  (in function itself we should use _.map instead of foreach/push).  i would be comfortable not testing this function